sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -16,33 +16,140 @@
 
 from copy import deepcopy
 import time
-from typing import Any, Optional
+from typing import Any, Awaitable, Callable, cast, Optional, overload, TypeVar, Union
```

Comment:
typing.Awaitable was deprecated in 3.9. We should do a follow-up and import this from abc if available...

Reply:
I see -- this was borrowed from https://github.com/Azure/azure-sdk-for-python/blob/55632a7ef295a48c9b2165d4341e7d0da654c1de/sdk/core/azure-core/azure/core/pipeline/policies/_authentication_async.py#L8

If this will cause issues, I can change the import source in a follow-up

Reply:
`Awaitable` is now imported from `collections.abc` here: https://github.com/Azure/azure-sdk-for-python/pull/37945/commits/c29dca4f0e2f62798e87d4b9c57810f52e611ecd